### PR TITLE
Migration guide for v6

### PR DESCRIPTION
**Pull request by stevengill**
###  Summary

Added a migration guide for the next majors coming out

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: docs/_tutorials/migrating_to_v6.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        return await database.get(InstallQuery.teamId);
+      }
+      throw new Error('Failed fetching installation');
+    },
+  },
+```
+
+### Deprecating @slack/events-api & @slack/interactive-messages
+
+`@slack/events-api` and `@slack/interactive-messages` will be deprecated on **January 12th, 2021**. We will only implement **critical bug fixes** until the official end of life date and close non critical issues and pull requests, which is slated for **May 31st, 2021**. At this time, development will fully stop for these packages and all remaining open issues and pull requests will be closed. Bolt for JavaScript can now perform all the same functionality as these packages. We think you’ll love the more modern set of features you get in Bolt. Here are some migration examples:
```
Comment (aoberoi) [inline docs/_tutorials/migrating_to_v6.md:91]:
```suggestion
`@slack/events-api` and `@slack/interactive-messages` will be deprecated on **January 12th, 2021**. We will only implement **critical bug fixes** until the official end of life date and close non critical issues and pull requests, which is slated for **May 31st, 2021**. At this time, development will fully stop for these packages and all remaining open issues and pull requests will be closed. [Bolt for JavaScript](https://slack.dev/bolt-js) can now perform all the same functionality as these packages. We think you’ll love the more modern set of features you get in Bolt. Here are some migration examples:
```

File: docs/_tutorials/migrating_to_v6.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+Before:
+```javascript
+// @slack/events-api
+slackEvents.on('app_home_opened', (event) -> {
+// fired when `app_home_opened` event is received
+// do work
```
Comment (aoberoi) [inline docs/_tutorials/migrating_to_v6.md:98]:
```suggestion
  // fired when `app_home_opened` event is received
  // do work
```

File: docs/_tutorials/migrating_to_v6.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+});
+
+// @slack/interactive-messages
+slackInteractions.action({actionId: 'buttonActionId'}, (payload, respond) => {
+// fired when a clicked button's actionId matches
+// do work
+respond();
```
Comment (aoberoi) [inline docs/_tutorials/migrating_to_v6.md:105]:
```suggestion
  // fired when a clicked button's actionId matches
  // do work
  respond();
```

File: docs/_tutorials/migrating_to_v6.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+});
+```
+
+After:
+```javascript
+// @slack/bolt
+app.event('app_home_opened', async ({ event }) -> {
+// fired when `app_home_opened` event is received
+// do work
```
Comment (aoberoi) [inline docs/_tutorials/migrating_to_v6.md:114]:
```suggestion
  // fired when `app_home_opened` event is received
  // do work
```

File: docs/_tutorials/migrating_to_v6.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+});
+
+// @slack/bolt
+app.action('buttonActionId', async( {action, ack} ) => {
+// fired when a clicked button's actionId matches
+// do work
+await ack();
```
Comment (aoberoi) [inline docs/_tutorials/migrating_to_v6.md:121]:
```suggestion
  // fired when a clicked button's actionId matches
  // do work
  await ack();
```

File: docs/_tutorials/migrating_to_v6.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+After:
+```javascript
+// @slack/bolt
+app.event('app_home_opened', async ({ event }) -> {
+// fired when `app_home_opened` event is received
+// do work
+});
+
+// @slack/bolt
+app.action('buttonActionId', async( {action, ack} ) => {
```
Comment (aoberoi) [inline docs/_tutorials/migrating_to_v6.md:118]:
```suggestion
app.action('buttonActionId', async({ action, ack }) => {
```

**Review by aoberoi — CHANGES_REQUESTED** [review]

File: docs/_tutorials/migrating_to_v6.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+After:
+```javascript
+// @slack/bolt
+app.event('app_home_opened', async ({ event }) -> {
+  // fired when `app_home_opened` event is received
+  // do work
+});
+
+// @slack/bolt
+app.action('buttonActionId', async({ action, ack }) => {
```
Comment (seratch) [inline docs/_tutorials/migrating_to_v6.md:118]:
```suggestion
app.action('buttonActionId', async ({ action, ack }) => {
```

File: docs/_tutorials/migrating_to_v6.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
+---
+title: Migrating to v6.x
+order: 1
+slug: migrating_to_v6
+---
+
+# Migration Guide (v5 to v6)
+
+The following packages have been updated in the release being dubbed as `v6`:
```
Comment (aoberoi) [inline docs/_tutorials/migrating_to_v6.md:9]: 👨‍🍳 💋

**Review by aoberoi — APPROVED** [review]

**Pull request merged**